osgar/drivers/rosmsg.py

```diff
@@ -80,12 +82,26 @@ def parse_raw_image(data, dump_filename=None):
     encoding = data[pos:pos+encoding_size]
     pos += encoding_size
 #    print(height, width, encoding)
+    assert encoding == b'32FC1', encoding
     is_bigendian, step, image_arr_size = struct.unpack_from('<BII', data, pos)
     pos += 1 + 4 + 4
+    # TODO refactor this part to use numpy, probably
+    # depth is array of floats
+    # slow: arr = [x[0] for x in struct.iter_unpack('f', data[pos:pos + image_arr_size])]
+    # cut min & max (-inf and inf are used for clipping) - use 16bit for test now
+    # slow: arr = [int(min(0xFFFF, max(0, x*1000))) for x in arr]
+    arr = np.frombuffer(data[pos:pos + image_arr_size], dtype=np.dtype('f'))*1000
+    arr = np.clip(arr, 0, 0xFFFF)
+    arr = np.ndarray.astype(arr, dtype=np.dtype('H'))
```

Comment:
It already uses `numpy` so the TODO is obsolete?

Why is the float array converted to 16bit int array? Why not keep the float array?

Reply:
The uint16 has half size and even 1mm resolution is waste. I would prefer to use byte per pixel but it is still not clear what we need for SubT. Note, that also packed unit16 is smaller.

Reply:
What about the part about the TODO? Is it obsolete?